### DISCUSSION TITLE
How @Value annotation works internally.

### DISCUSSION THREAD

**Discussion by arvindersinghchhabra** [junit-team/junit-framework#2799] [Q&A]
How @Value annotation works internally in spring.

Actually I want to implement the same using custom annotation by providing logic and assign back value to variable field again.

In spring we do

@Value(${key})
private String name;

But I want

@CustomAnnotation
private String name;

Here @CustomAnnotation will read value of name variable and perform some logic on name variable value and assign back to name variable.

Appreciate your help !!!!

**Comment by marcphilipp** [comment]
@arvindersinghchhabra This sounds like a Spring, not a JUnit question to me. I think you'll have better luck to get an answer by asking on StackOverflow.
